io/zenoh-transport/src/unicast/shm/transport.rs

```diff
@@ -0,0 +1,329 @@
+//
+// Copyright (c) 2023 ZettaScale Technology
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+// which is available at https://www.apache.org/licenses/LICENSE-2.0.
+//
+// SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+//
+// Contributors:
+//   ZettaScale Zenoh Team, <zenoh@zettascale.tech>
+//
+use super::link::send_with_link;
+use super::oam_extensions::pack_oam_close;
+#[cfg(feature = "stats")]
+use super::TransportUnicastStatsAtomic;
+use crate::transport_unicast_inner::TransportUnicastInnerTrait;
+use crate::TransportConfigUnicast;
+use crate::TransportManager;
+use crate::{TransportExecutor, TransportPeerEventHandler};
+use async_executor::Task;
+#[cfg(feature = "transport_shm")]
+use async_std::sync::RwLockUpgradableReadGuard;
+use async_std::sync::{
+    Mutex as AsyncMutex, MutexGuard as AsyncMutexGuard, RwLock, RwLockReadGuard,
+};
+use async_std::task::JoinHandle;
+use async_trait::async_trait;
+use std::ops::Deref;
+use std::sync::{Arc, RwLock as SyncRwLock};
+use std::time::Duration;
+#[cfg(feature = "transport_shm")]
+use zenoh_core::zasyncread_upgradable;
+use zenoh_core::{zasynclock, zasyncread, zread, zwrite};
+use zenoh_link::{Link, LinkUnicast, LinkUnicastDirection};
+use zenoh_protocol::core::{WhatAmI, ZenohId};
+use zenoh_protocol::network::NetworkMessage;
+use zenoh_protocol::transport::{Close, TransportSn};
+use zenoh_result::{bail, zerror, ZResult};
+
+/*************************************/
+/*             TRANSPORT             */
+/*************************************/
+#[derive(Clone)]
+pub(crate) struct ShmTransportUnicastInner {
```

Comment:
For naming consistency I'd call it `TransportUnicastShm` and `TransportUnicastNet`, both implement `TransportUnicastTrait`. Opinion?